iXBRLViewerPlugin/viewer/src/js/util.js

```diff
@@ -13,25 +13,38 @@
 // limitations under the License.
 
 import dateFormat from "dateformat"
+import moment from "moment";
 
-export function isodateToHuman(d, adjust) {
-    if (d.getUTCHours() + d.getUTCMinutes() + d.getUTCSeconds() == 0) { 
+/* 
+ * Takes a moment.js oject and converts it to a human readable date, or date
+ * and time if the time component is not midnight.  Adjust specifies that a
+ * date (but not a date time) should be shown as the day before.  This is to
+ * satisfy the convention of describing durations using inclusive dates.
```

Comment:
Describing it in terms of "inclusive" seems odd to me since the adjustment is needed because duration end dates are exclusive (and for instances, which aren't inclusive or exclusive).  Perhaps "This is to satisfy the convention of describing 24:00:00 as 00:00:00 of the following day for instants and duration end dates."

Reply:
Internally, the viewer works with full date times, so an end date for one period and the start date for the next are the same time.  This comment is referring to the fact that the "human" way to describe the year from 2018-01-01T00:00:00 to 2019-01-01T00:00:00 is "1st Jan 2018 to 31st Dec 2018" i.e. the first and last days included in the period (with the side effect that we describe the end of one period and the beginning of the next differently even thought they're the same point in time).  I've expanded the comment in 395114d0d946973bad5e0381a5d6d13fb3a778ff .

Reply:
Ah, my misunderstanding.  Thanks for clarifying.